### DISCUSSION TITLE
Make tauri compatible with .NET (interop C#/VB with Rust)

### DISCUSSION THREAD

**Discussion by RubenPX** [tauri-apps/tauri#9114] [Show and tell]
Recently i tried to make a rust compatible with C# (just for fun)

Using [netcorehost](https://github.com/OpenByteDev/netcorehost) by @OpenByteDev

It's a hybrid that supports C# and Rust with Tauri.

The project name is [TauriNET ](https://github.com/RubenPX/TauriNET) (I don't know if i put this name in repository it violate any rule or something)

I don't know if this project is useful for something, but i liked to make it, and it's super small code

Link here: https://github.com/RubenPX/TauriNET

**Comment by tats-u** [comment]
It sounds great but it's a pity that the license is not permissive one, MPL, or at least LGPL but GPL.

**Comment by cornzzy** [comment]
The industry standard for sharing code between languages is [shared library](https://en.wikipedia.org/wiki/Shared_library) such as .so (android), .xcframework (ios), .dll (windows) and so on. Export dll from your dot net and life will be easier.

**Reply by tats-u** [reply]
Software that links dynamic libraries licensed under GPL (not LGPL) must be licensed under GPL or AGPL, too.
If the authors of dynamic libraries don't want to force users to license their software under GPL, they must choose LGPL or MPL instead.

**Reply by RubenPX** [reply]
Wow. I didn't know about shared library. Sounds easy for developers. Y will try

**Reply by tats-u** [reply]
In .NET, depended packages are bundled as DLLs (shard libraries) unless a user chooses `PublishSingleFile`.
If you don't require software authors who just use your library to expose the source code, you might want to choose one of the followings instad of GPL:

- LGPL (If you don't want authors of software that uses your library to stop their users to do reverse engineering + MPL)
- MPL (If just you want *authors of modified versions of your library* to expose their source code)
- A permissive license (e.g. BSD, MIT) (If you just want your users to *expose the fact that they are using your library*)

**Reply by RubenPX** [reply]
I choose MIT License

**Reply by tats-u** [reply]
You don't require modified libraries (possibly proprietary) to publish the difference.

**Reply by RubenPX** [reply]
> You don't require modified libraries (possibly proprietary) to publish the difference.

I think i don't use modified libraries.

In this case are all crate modules

**Reply by tats-u** [reply]
FYI there are some people who want to merge every beneficial change in modified versions including proprietary ones. 
Anyway I'll respect your decision.
